src/node.cc

```diff
@@ -2868,6 +2871,13 @@ static bool ExecuteBootstrapper(Environment* env, Local<Function> bootstrapper,
 
 
 void LoadEnvironment(Environment* env) {
+  if (standalone_mode) {
+    env->isolate()->AddMessageListener(OnMessage);
```

Comment:
To make things more adventurous, one `Start` calls the other, just so you know. 😅

Reply:
> ... Can you explain?

Yeah, I'm with @bnoordhuis on this... it's not obvious why they're separate.

Reply:
@bnoordhuis @jasnell I think they're separate because electron needs both the cases (one for the main process and another for the renderer).

Quoting @codebytere here:

> basically, we have 3 modes when running Node in Electron:  
> 1. In browser process, we want to have a full Node environment, but with signal handlers and other low level things disabled.
> 2. In renderer process, we want Node to reuse web page's context.
> 3. In `ELECTRON_RUN_AS_NODE`, we want Node to run like the official one. The combination of the 2 flags controls in which mode Node should run

Hope that clears everything?

Reply:
@ryzokuken covered it essentially but the net set of needs mean that we can't just have one flag and need two, as there are three total mode sets

Reply:
> Hope that clears everything?

No, because I'd already read that. :-)

I interpret "reuse web page's context" as meaning there's an existing v8::Context that Node.js should use but then why install per-isolate hooks?

I guess it's not entirely clear to me how the second mode works or what the life cycle of that context is.

Reply:
@bnoordhuis sorry, I didn't read the PR body 😅

Reply:
> I interpret "reuse web page's context" as meaning there's an existing v8::Context that Node.js should use but then why install per-isolate hooks?
> 
> I guess it's not entirely clear to me how the second mode works or what the life cycle of that context is.

In the second mode, Node is inserted in web pages by:

1. blink creates `v8::Context`;
2. we create `node::Environment` on the context of blink;
3. blink notifies it is going to destroy v8 context;
4. we destroy the node environment;
5. blink destroy the v8 context.

For the renderer process, we disable all global hooks in Node (like signal handlers and isolate hooks) by setting both `standalone_mode` and `upstream_node_mode` to `false`.